### PR TITLE
FEATURE: add v2 indicator set api

### PR DESCRIPTION
**Pull request by c9s**

### CODE REVIEW

**Review by gx578007 — APPROVED** [review]

File: pkg/indicator/klinestream.go [hunk 1]
```diff
@@ -36,6 +36,10 @@ func (s *KLineStream) AddSubscriber(f func(k types.KLine)) {
 	}
 }
 
+func (s *KLineStream) AddBackLog(kLines []types.KLine) {
```
Comment (zenixls2) [inline pkg/indicator/klinestream.go:39]: How about rename this to BackfillKLines?

File: pkg/bbgo/indicator_loader.go [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+func (i *IndicatorSet) CLOSE(interval types.Interval) *indicator.PriceStream {
+	if closePrices, ok := i.closePrices[interval]; ok {
+		return closePrices
+	}
+
+	closePrices := indicator.ClosePrices(i.KLines(interval))
+	i.closePrices[interval] = closePrices
+	return closePrices
+}
+
```
[the remark this answers is not in the document]
Comment (c9s) [reply pkg/bbgo/indicator_loader.go:67]: sure

File: pkg/bbgo/indicator_loader.go [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+func (i *IndicatorSet) KLines(interval types.Interval) *indicator.KLineStream {
+	if kLines, ok := i.kLines[interval]; ok {
+		return kLines
+	}
+
+	kLines := indicator.KLines(i.stream, i.Symbol, interval)
+	if kLinesWindow, ok := i.store.KLinesOfInterval(interval); ok {
+		kLines.AddBackLog(*kLinesWindow)
+	}
```
[the remark this answers is not in the document]
Comment (c9s) [reply pkg/bbgo/indicator_loader.go:40]: okay

File: pkg/indicator/klinestream.go (hunk 1 above)
Comment (c9s) [reply pkg/indicator/klinestream.go:39]: Or maybe just Backfill?

Comment (zenixls2) [reply pkg/indicator/klinestream.go:39]: up to you

File: pkg/bbgo/indicator_loader.go (hunk 3 above)
[the remark this answers is not in the document]
Comment (c9s) [reply pkg/bbgo/indicator_loader.go:44]: done

File: pkg/bbgo/indicator_loader.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (c9s) [reply pkg/bbgo/indicator_loader.go:71]: done

**Review by zenixls2 — APPROVED** [review]

**Pull request merged**